### PR TITLE
Don't die when we run out of randomness

### PR DESCRIPTION
**Pull request by mollyling**
I'm often hitting out-of-randomness issues (when running uProxy/rappor-test-client.git).  This smoothes over it.  Better solutions are certainly possible, but this is a backstop to avoid halting.

### CODE REVIEW

File: src/crypto.js [hunk 1]
```diff
@@ -40,12 +40,10 @@ exports.getRandomValues = function (buffer) {
     view = new Uint8Array(buffer),
     i;
 
-  if (offset + size > buf.length) {
-    throw new Error("Insufficient Randomness Allocated.");
-  }
-
+  // Hack: just wrap around.
```
Comment (willscott) [inline src/crypto.js:43]: Reusing randomness is bad news bears. We need to figure out what part of algorithm is using more bytes of randomness than we expect, and instead request the correct amount of entropy from the system.
This approach absolutely opens the algorithm to confirmation / information disclosure attacks.

Comment (mollyling) [inline src/crypto.js:43]: Yup, fully agreed.  But we're not generating enough randomness sometimes, and failing isn't helpful.  So, an alternative here would be to adjust this to be a promise, and let it invoke refreshBuffer() as needed. WDYT?

Comment (willscott) [inline src/crypto.js:43]: i'd really like a design where it's hard to shoot yourself in the foot. Having to check something afterward to know that you ran out crypto and should re-run with a bigger buffer seems like it's too easy to not do that check.

A design I think would remain safe is a switch to put crypto into a 'trial run' mode, where it gives back zeros and counts how many bytes are requested. then you can run once to learn the buffer size, and then again with entropy allocated. would that work?

Comment (mollyling) [inline src/crypto.js:43]: I think that would unnecessarily add hidden to this system (which is there now, but should get removed).

When would you know when the trial ends?

I found this while writing https://github.com/uProxy/rappor-test-client/blob/master/test-client.js, which creates an array of metrics and aggregates them into a single JSON (simulating multiple senders of the metrics).  I don't know when you'd be able to tell when the client is done queueing all the promises for metric.report(..) before you'd get enough information on how much random entropy you need.

Comment (willscott) [inline src/crypto.js:43]: what do you mean 'add hidden'?

Comment (mollyling) [inline src/crypto.js:43]: Ugh, too much editing.  I meant add hidden state -- the amount of entropy that this component believes that it needs.

**Comment by mollyling** [comment]
Ref: #12.  Can we fix this, please?  I present that you can't know how much entropy you need beforehand.

**Comment by willscott** [comment]
We were previously able to know how much entropy ahead of time. what's changed? Rappor is extremely deterministic in how many bytes of randomness it calls for - we're just using new constructs like strings where we don't have the buffer sizes calculated correctly.

**Comment by mollyling** [comment]
Ok, I didn't notice that new call to getRandomness every time we generated a report.  I thought it assumed a pre-set number of report generations.  I've got another PR (#14) to fix the bugs it was hitting in calculating and getting the randomness.

**Pull request closed without merging**